app/services/katello/registration_manager.rb

```diff
@@ -79,6 +79,9 @@ def validate_hosts(hosts, organization, host_name, host_uuid, host_uuid_overridd
           if host.name == host_name
             unless Setting[:host_profile_assume] || host.subscription_facet.nil? || host.build || host_uuid_overridden
               current_dmi_uuid = host.subscription_facet.dmi_uuid
+              if not Setting[:host_re_register] and not host.build and current_dmi_uuid
```

Comment:
there's some rubocop errors:

 Use ! instead of not.
Use && instead of and.

Reply:
I'll update and re-push.